internals/cli/cmd_okay.go

```diff
@@ -61,33 +61,28 @@ func (cmd *cmdOkay) Execute(args []string) error {
 		return ErrExtraArgs
 	}
 
+	state, err := loadCLIState(cmd.socketPath)
+	if err != nil {
+		return fmt.Errorf("cannot load CLI state: %w", err)
+	}
+
+	okayedWarnings := false
+	if !state.WarningsLastListed.IsZero() {
+		okayedWarnings = true
+		state.WarningsLastOkayed = state.WarningsLastListed
+	}
+
 	okayedNotices := false
 	if !cmd.Warnings {
-		state, err := loadCLIState(cmd.socketPath)
-		if err != nil {
-			return fmt.Errorf("cannot load CLI state: %w", err)
-		}
 		if !state.NoticesLastListed.IsZero() {
 			okayedNotices = true
 			state.NoticesLastOkayed = state.NoticesLastListed
-			err = saveCLIState(cmd.socketPath, state)
-			if err != nil {
-				return fmt.Errorf("cannot save CLI state: %w", err)
-			}
 		}
 	}
 
-	last, err := lastWarningTimestamp()
+	err = saveCLIState(cmd.socketPath, state)
```

Comment:
I think, at some point, we need to improve `saveCLIState` to atomically write the state file (and to include conflict resolution).